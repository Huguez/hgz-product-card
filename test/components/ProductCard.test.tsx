import React from "react"
import renderer, { act } from "react-test-renderer"
import { ProductCard } from '../../src/components';
import { product } from '../data/data';


describe( "ProductCard", ()=>{

   test( "Debe de renderizarse", () => {
      const wrapper = renderer.create( 
         <ProductCard product={ product } >
            {
               () => (
                   <h1>Hola wawa</h1>
               )
            }
         </ProductCard>
      )
      expect( wrapper.toJSON() ).toMatchSnapshot();
   } );

   test( "", () => {
      const wrapper = renderer.create( 
         <ProductCard product={ product } >
            {
               ({ count, increaseBy }) => (
                  <>
                   <h1>{ count }</h1>
                   <button onClick={ () => increaseBy(1) }>+1</button>
                  </>
               )
            }
         </ProductCard>
      )
      let tree = wrapper.toJSON()

      expect( tree ).toMatchSnapshot();
      
      act( ()=>{
         (tree as any).children[1].props.onClick()
      } )
      
      tree = wrapper.toJSON()
      
      expect( (tree as any).children[0].children[0] ).toBe("1")
   } );

} );
