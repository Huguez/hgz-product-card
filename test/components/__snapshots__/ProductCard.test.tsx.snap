// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard  1`] = `
<div
  className="undefined  productCard "
>
  <h1>
    0
  </h1>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</div>
`;

exports[`ProductCard Debe de renderizarse 1`] = `
<div
  className="undefined  productCard "
>
  <h1>
    Hola wawa
  </h1>
</div>
`;
